app/utils/qrdisplay.js

```diff
@@ -12,5 +12,9 @@ export const createQrImage = (elem, data, size) => {
     const oldChild = elem.firstElementChild
     if (oldChild) { oldChild.remove() }
     elem.appendChild(image)
+  } else if (!data && elem) {
+    const errorMessage = document.createElement('p')
+    errorMessage.innerHTML = "Error loading QR code. Please check your internet connection and try again."
```

Comment:
We should make this a localised message maybe `CHECK_INTERNET_CONNECTION` for now ... and combine it with `RETRY` so that we don't need new strings 😛